### PR TITLE
Remove /health and /ready from Access Logs

### PR DESCRIPTION
**Pull request by renuka-fernando**
### Purpose
$subject

Config
```yaml
[accessLogs.excludes]
[accessLogs.excludes.systemHost]
enabled = true
pathRegex = "^(/health|/ready)$" # keep this empty to match all paths
```

#### 1. System Host /health
```sh
$ curl -X GET "https://localhost:9095/health" -H "accept: application/json" -H "Authorization:Bearer $TOKEN" -k
{"status": "healthy"}
```

No Access log printed.

#### 2. Other Host /health

```sh
$ curl -X GET "https://localhost:9095/health" -H "accept: application/json" -H "Authorization:Bearer $TOKEN" -k -H 'Host: foo'
{"description":"The requested resource is not available.","code":"404","message":"Not Found"}
```

```json
{"apiPath":null,"respTxDur":null,"respCode":404,"upstrmSvcTime":null,"respCodeDtls":"route_not_found","respDur":null,"bytesSent":94,"extAuthDtls":null,"dur":0,"respFlag":"NR","time":"2023-09-04T05:48:03.697Z","reqTxDur":null,"prot":"HTTP/1.1","upstrmHost":null,"bytesRecv":0,"reqDur":0,"ua":"curl/7.79.1","reqId":"b91f7d9b-8eb4-438c-8fed-eacb227011a8","method":"GET","gwHost":null,"xff":null,"host":"foo","apiUuid":null,"upstrmPath":"/health"}
```

#### 3. System Host /health/foo/health

```sh
$ curl -X GET "https://localhost:9095/health/foo/health" -H "accept: application/json" -H "Authorization:Bearer $TOKEN" -k
{"description":"The requested resource is not available.","message":"Not Found","code":"404"}
```

```json
{"respCodeDtls":"route_not_found","apiPath":null,"time":"2023-09-04T05:48:52.486Z","upstrmSvcTime":null,"reqDur":0,"respDur":null,"xff":null,"ua":"curl/7.79.1","bytesRecv":0,"extAuthDtls":null,"respFlag":"NR","reqId":"cead9a49-9fe7-4ebd-8895-f3804ca1237b","dur":0,"gwHost":null,"bytesSent":94,"respCode":404,"upstrmPath":"/health/foo/health","host":"localhost:9095","respTxDur":null,"reqTxDur":null,"upstrmHost":null,"prot":"HTTP/1.1","method":"GET","apiUuid":null}
```

#### 4. System Host /token

```sh
curl -X POST "https://localhost:9095/testkey" -d "scope=read:pets" -H "Authorization: Basic YWRtaW46YWRtaW4=" -k -v
```

```json
{"upstrmSvcTime":null,"respTxDur":0,"extAuthDtls":null,"reqId":"22b2a953-e0ba-40f6-86a1-e28d9e97a7ee","prot":"HTTP/1.1","method":"POST","bytesRecv":15,"apiUuid":null,"bytesSent":762,"xff":null,"ua":"curl/7.79.1","respFlag":"-","respCodeDtls":"via_upstream","host":"enforcer","upstrmPath":"/testkey","reqDur":0,"respDur":108,"gwHost":null,"upstrmHost":"172.30.0.4:9001","dur":108,"apiPath":null,"time":"2023-09-04T06:07:49.854Z","reqTxDur":36,"respCode":200}
```

#### 5. Invoke API

```sh
curl -X GET "https://localhost:9095/v2/pet/findByStatus?status=pending" -H "accept: application/json" -H "Authorization:Bearer $TOKEN" -k
```

```json
{"dur":2006,"bytesRecv":0,"upstrmHost":"185.42.117.109:443","reqId":"781d3001-7d47-4989-a2fa-767f095c1229","respFlag":"-","reqDur":0,"upstrmPath":"/v3/9ec5a57e-733c-4338-9854-7ea31866dcba/pet/findByStatus?status=pending","extAuthDtls":null,"reqTxDur":1845,"respCodeDtls":"via_upstream","respDur":2005,"respCode":200,"prot":"HTTP/1.1","bytesSent":31,"apiPath":"/v2/pet/findByStatus?status=pending","gwHost":"localhost:9095","ua":"curl/7.79.1","apiUuid":"648957e7a675c35145393369","upstrmSvcTime":null,"method":"GET","host":"run.mocky.io","time":"2023-09-04T06:08:54.093Z","xff":null,"respTxDur":0}
```

### Issues

Fixes #

### Automation tests
 - Unit tests added: N/A
 - Integration tests added: N/A

### Tested environments

Tested in Docker Compose

---
#### Maintainers: Check before merge
- [x] Assigned 'Type' label
- [x] Assigned the project
- [x] Validated respective github issues
- [x] Assigned milestone to the github issue(s)

### CODE REVIEW

File: adapter/internal/oasparser/envoyconf/access_loggers.go [hunk 1]
```diff
@@ -168,3 +172,68 @@ func getAccessLogs() []*config_access_logv3.AccessLog {
...
+func getAccessLogFilterConfig() *config_access_logv3.AccessLogFilter {
+	logConf := config.ReadLogConfigs()
+	conf, _ := config.ReadConfigs()
+
+	if logConf.AccessLogs.Excludes.SystemHost.Enabled {
+		logger.LoggerOasparser.Debugf("Access log excludes for system host is enabled with path regex: %q",
+			logConf.AccessLogs.Excludes.SystemHost.PathRegex)
+	} else {
+		return nil
+	}
```
Comment (slahirucd7) [inline adapter/internal/oasparser/envoyconf/access_loggers.go:186]: Do we need this else part ?

File: adapter/internal/oasparser/envoyconf/access_loggers.go [hunk 2]
```diff
@@ -168,3 +172,68 @@ func getAccessLogs() []*config_access_logv3.AccessLog {
...
+
```
Comment (slahirucd7) [inline adapter/internal/oasparser/envoyconf/access_loggers.go:187]:
```suggestion

	if !logConf.AccessLogs.Excludes.SystemHost.Enabled {
		return nil
	} 
	logger.LoggerOasparser.Debugf("Access log excludes for system host is enabled with path regex: %q",
			logConf.AccessLogs.Excludes.SystemHost.PathRegex)

```

File: resources/conf/log_config.toml [hunk 3]
```diff
@@ -29,6 +29,11 @@ enable = false
 logfile = "/dev/stdout" # This file will be created inside router container.
 logType = "text" # LogTypes can be "text" or "json"
 
+[accessLogs.excludes]
```
Comment (slahirucd7) [inline resources/conf/log_config.toml:32]: Shall we add this to the templates file too ?

File: resources/conf/log_config.toml [hunk 4]
```diff
@@ -67,3 +71,5 @@ reqTxDur = "%REQUEST_TX_DURATION%"
 respTxDur = "%RESPONSE_TX_DURATION%"
 reqDur = "%REQUEST_DURATION%"
 respDur = "%RESPONSE_DURATION%"
+apiUUID = "%DYNAMIC_METADATA(envoy.filters.http.ext_authz:apiUUID)%"
```
Comment (slahirucd7) [inline resources/conf/log_config.toml:74]: Isn't it better to keep API UUID in the beginning ?

**Review by NomadXD — APPROVED** [review]

File: resources/conf/log_config.toml (hunk 3 above)
Comment (renuka-fernando) [reply resources/conf/log_config.toml:32]: For log_config.toml we don't have a template file

File: resources/conf/log_config.toml (hunk 4 above)
Comment (renuka-fernando) [reply resources/conf/log_config.toml:74]: Yeah, it is better. But the reason I changed the order was, in TEXT format we have appended the apiUUID at the end of the line. To have the same order I moved ApiUUID down.

File: adapter/internal/oasparser/envoyconf/access_loggers.go (hunk 2 above)
Comment (renuka-fernando) [reply adapter/internal/oasparser/envoyconf/access_loggers.go]: Fixed with https://github.com/wso2/product-microgateway/commit/99e77ecae08191a5846fcc348827c032d7c89e45

**Review by NomadXD — APPROVED** [review]

File: resources/conf/log_config.toml (hunk 3 above)
Comment (slahirucd7) [reply resources/conf/log_config.toml:32]: Let's add a comment then similar to the 
https://github.com/wso2/product-microgateway/pull/3430/files#diff-757ae11db4f6dc0be3215293e11e52484ab29526d0a15f884fc3f67f50630730L25
and
https://github.com/wso2/product-microgateway/pull/3430/files#diff-757ae11db4f6dc0be3215293e11e52484ab29526d0a15f884fc3f67f50630730L14

With another PR.

**Review by slahirucd7 — APPROVED** [review]

**Comment by choreo-cicd** [comment]
[succeeded] Dataplane(EastUS) cluster : dev-deployment-v2 : 20230905.39

**Comment by choreo-cicd** [comment]
[succeeded] Dataplane(NorthEU) cluster : dev-deployment-v2 : 20230905.39

**Comment by choreo-cicd** [comment]
[succeeded] Controlplane cluster : dev-deployment-v2 : 20230905.39

**Comment by choreo-cicd** [comment]
[succeeded] Dataplane(NorthEU) cluster : prod-deployment-v2 : 20230907.5

**Comment by choreo-cicd** [comment]
[succeeded] Controlplane cluster : prod-deployment-v2 : 20230907.5

**Pull request merged**